### PR TITLE
Upgrade to Gradle 5.0 

### PR DESCRIPTION
**Pull request by alpar-t**
- Upgrade to Greadle 5.0 
- updates to comply with breaking changes in Groovy. 
It seems like the latest version brought along by Gradle changes how the `final` keyword is interpreted, 
unfortunately it can't see past branches so it doesn't work like ti would in java.

This is a work in progress to get the tests running. Will only merge when Gradle 5.0 is GA.

Release Notes: https://github.com/gradle/gradle/releases/tag/v5.0.0

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: why is final removed?

**Review by rjernst — COMMENTED** [review]
I have one question, but I also don't think we should upgrade to a milestone release. This is like upgrading to a beta.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
Comment (atorok) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: I will not merge the PR until there is a GA, just wanted to start running in CI. 

Gradle bumps the Groovy version, which now complains that the final variable is changed. 
It doesn't see past branches the same way java does. One more reason to switch to java.

**Comment by atorok** [comment]
@elasticmachine test this please

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/RestIntegTestTask.groovy
Comment (atorok) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/RestIntegTestTask.groovy]: The static method was called with the value of `includePackagedTests` in what happened to be the right order so that changes to the property were picked up in time and the right value was passed into the `project.afterEvaluate` ( which had no use, since the value was already picked up anyhow ). 
This can't be static to make changes to the property reflected as they should. Gradle 5 seems to have changed the order in which some operations are executed.  I wish they would have an option to randomize ordering of otherwise undefined sequences.

**Comment by atorok** [comment]
@rjernst Gradle 5.0 is GA and this is ready for review and merge now.

**Comment by droberts195** [comment]
One thing to note is that in release manager Gradle will run with ml-cpp and found-elasticsearch-plugin in the elasticsearch-extra directory.  Were there any particular things that needed changing as part of this PR that are also likely to need changing in the `build.gradle` files in those repos before Gradle 5 will work with them?  I'm happy to make the ml-cpp changes if any are required, but it would be good to consider this before merging this PR because otherwise the unified snapshot will be broken in the meantime.

**Comment by atorok** [comment]
@droberts195 The best thing is to run with a build scan and look at deprecation messages. 
The `finally` keyword changed how it works and I had to remove it from a bunch of places. 
Other than that this was not special  from minor version updates where things are ran in different order and this uncovers some explicit ordering that is missing.  I'm also glad to help making those changes. I think the hadoop plugin is also affected ( /cc @jbaiera )

**Comment by droberts195** [comment]
I can confirm that ml-cpp now works with Gradle 5

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by atorok** [comment]
`gradlew` in the found plugin actually uses the one from the ES repo. I checked to make sure Gradle 5 doesn't cause any issues there.

**Pull request merged**
